Code/app/src/main/java/ca/ualberta/moodroid/ui/EditDeleteFragment.java

```diff
@@ -40,15 +67,38 @@ public void onClick(DialogInterface dialogInterface, int i) {
                 .setPositiveButton("Edit", new DialogInterface.OnClickListener() {
                     @Override
                     public void onClick(DialogInterface dialogInterface, int i) {
-                        // ADD EDIT VIEW
+                        //gets the mood event by id
+                        Intent intent = new Intent(getActivity(), EditMoodDetail.class);
+                        intent.putExtra("eventId", id);
+                        intent.putExtra("emoji", bundle.getString("emoji"));
+                        intent.putExtra("mood_name", bundle.getString("mood_name"));
+                        intent.putExtra("hex", bundle.getString("hex"));
+                        startActivity(intent);
                     }
                 })
                 .setNeutralButton("Delete", new DialogInterface.OnClickListener() {
                     @Override
                     public void onClick(DialogInterface dialogInterface, int i) {
-                        // DELETE METHOD
+                        MoodEventService moodEvent = new MoodEventService();
+                        //gets the mood event by id
+                        moodEvent.getEventWithId(id).addOnSuccessListener(new OnSuccessListener<MoodEventModel>() {
+                            @Override
+                            public void onSuccess(MoodEventModel moodEventModel) {
+                                //For some reason internalid is set to null
+                                moodEventModel.setInternalId(id);
+                                MoodEventRepository mood = new MoodEventRepository();
+                                mood.delete(moodEventModel).addOnSuccessListener(new OnSuccessListener<Void>() {
+                                    @Override
+                                    public void onSuccess(Void aVoid) {
+                                        Log.d("RESULT/DELETE", "Model Deleted");
+                                        //TODO: currently this causes a crash
+                                        //((MoodHistory) getActivity()).getMood();
```

Comment:
Does this still not work?

Reply:
Maybe you can use http://simpledeveloper.com/how-to-communicate-between-fragments-and-activities/ something like this, implement a listener that the fragment can call. Just do some googling.